test/integration/api/forms/datasets.js

```diff
@@ -112,3 +112,106 @@ describe('api: /projects/:id/forms (entity-handling)', () => {
     }));
   });
 });
+
+describe('api: /projects/:id/forms/draft/dataset', () => {
```

Comment:
looks good - tests seem to cover all cases i can think of. 
actually, what happens on a form with no dataset defined? does it return [] or does it crash?